src/main/java/org/jenkinsci/plugins/badge/PublicBadgeAction.java

```diff
@@ -127,11 +128,11 @@ public HttpResponse doCoverageIcon(StaplerRequest req, StaplerResponse rsp, @Que
         Job<?, ?> project = getProject(job);
         Integer codeCoverage = null;
 
-        if (project.getLastSuccessfulBuild() != null) {
```

Comment:
Right, that is what I intended. The idea is to correlate all badges with the build status badge, so it is a true representation of the latest status.
Maybe a good middle ground is to have this behavior configurable through Jenkins?